### PR TITLE
Add unit tests for SampleAdaptiveOffset::offsetBlock

### PR DESCRIPTION
**Pull request by micro-arm**
Add unit tests for `SampleAdaptiveOffset::offsetBlock` by comparing the scalar and SIMD implementations across SAO types, block sizes, availability cases and virtual-boundary cases.

Add an `enableOpt` switch to the `SampleAdaptiveOffset` so the unit test can construct scalar and optimized instances for direct comparison.

Move the `offsetBlock` function pointer from protected to public so the unit test can invoke the selected scalar or SIMD implementation directly.

### CODE REVIEW

File: tests/vvdec_unit_test/vvdec_unit_test.cpp [hunk 1]
```diff
@@ -1697,6 +1698,268 @@ static bool test_RdCost()
...
+  SampleAdaptiveOffset opt{ /*enableOpt=*/true };
+
+  unsigned numCases = NUM_CASES;
+  bool passed = true;
+
+  for( int bitDepth : { 8, 10 } )
+  {
+    for( int typeIdx : { SAO_TYPE_EO_0, SAO_TYPE_EO_90, SAO_TYPE_EO_135, SAO_TYPE_EO_45, SAO_TYPE_BO } )
+    {
+      for( unsigned h : { 8, 16, 32, 48, 64, 128 } )
```
Comment (adamjw24) [inline tests/vvdec_unit_test/vvdec_unit_test.cpp:1949]: The block-sizes in this case are CTU-sizes. I.e. its 32x32, 64x64 or 128x128, +the clipping on the picture boundary (and the picture size can be any multiple of 8). I see you have the 48 in there, but its the only non-power-of-2 size in there. I would add some strange sizes in there for good measure, like 40, 120 or 72 (non-power-of-2 multiple of 8, but not 16, which is missing from the set for now).

File: tests/vvdec_unit_test/vvdec_unit_test.cpp [hunk 2]
```diff
@@ -1697,6 +1698,268 @@ static bool test_RdCost()
...
+      { 0, 2, SaoVirtualBoundaryCase::Middle,     "twoHorizontal"   },
+      { 2, 2, SaoVirtualBoundaryCase::Middle,     "twoEach"         },
+      { 2, 2, SaoVirtualBoundaryCase::NearEdge,   "twoEachNearEdge" },
+  };
+
+  bool passed = true;
+  constexpr unsigned allAvailCaseIdx = sizeof( availCases ) / sizeof( availCases[0] ) - 1;
+  const unsigned numAvailCases = typeIdx == SAO_TYPE_BO ? 1 : sizeof( availCases ) / sizeof( availCases[0] );
+  constexpr unsigned numVbCases = sizeof( vbCases ) / sizeof( vbCases[0] );
+  const unsigned numTestCases = std::max( numCases, numAvailCases * numVbCases );
```
Comment (adamjw24) [inline tests/vvdec_unit_test/vvdec_unit_test.cpp:1924]: For standard value of `NUM_CASES`, is all of the combinations getting tested or does the clipping always remove some cases?

Comment (adamjw24) [reply tests/vvdec_unit_test/vvdec_unit_test.cpp:1924]: Oh, sorry, thats a `max`, not a `min`, my bad. All good here

File: tests/vvdec_unit_test/vvdec_unit_test.cpp (hunk 1 above)
Comment (micro-arm) [reply tests/vvdec_unit_test/vvdec_unit_test.cpp:1949]: I can extend the width loop values like this: `w : { 8, 16, 24, 32, 40, 64, 72, 120, 128 } )`

But for the height I don't think it's necessary since it's the outer loop?

Comment (adamjw24) [reply tests/vvdec_unit_test/vvdec_unit_test.cpp:1949]: Sounds good to me

Comment (micro-arm) [reply tests/vvdec_unit_test/vvdec_unit_test.cpp:1949]: Thanks. I pushed the revised widths.
